File: cookbook/php/createValueList.php
<?php

//Ici, renseignez la xKey et les parametres personnalises du message
$xKey = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
$valueListsId = 012;	//Id de la liste a modifier ('null' si le critere est a creer)

$name = 'createValueList (php)';	//Nom de la liste
$ordered = false;	//Ordonnee : oui = 'true' / non = 'false'
$values = array(	//Valeurs de la liste
			array(	//Premiere valeur
				'index' => 1,	//Numeros de la valeur
				'value' => "Mr"),	//Valeur du champs
			array(	//Deuxieme valeur
				'index' => 2,	//Numeros de la valeur
				'value' => "Mme")	//Valeur du champs
			);

//On trouve l'addresse pour la requete
$url = 'http://v8.mailperformance.com/valueLists/' . $valueListsId;

//Creation du Json du message

$arr = array(
		'name' => $name,
		'ordered' => $ordered,
		'values' => $values);


//On affiche le message
$message = json_encode($arr);
echo $message . "\n";

//Connexion
if ($valueListsId != null)
	$con = connect($url, $xKey, $message, 'PUT');
else
	$con = connect($url, $xKey, $message, 'POST');

$result = $con['result'];
$info = $con['info'];
$req = $con['req'];

if ($info['http_code'] != 200)
{
	echo 'Error : ' . $info['http_code'];
}
else
{
	//La liste de valeurs a bien ete cree
	echo 'The value list : ' . $name . " is created / update.";
}
curl_close($req);



//Fonctions -----



//Utilisation de cURL pour remplir les requetes
function startCurlInit($url)
{
	$init = curl_init();
	curl_setopt($init, CURLOPT_URL, $url);
	curl_setopt($init, CURLOPT_RETURNTRANSFER, true);
	return ($init);
}

//Fonction de connexion
function connect($url, $xKey, $message, $method)
{
	//On remplit la requete
	$req = startCurlInit($url);
	curl_setopt($req, CURLOPT_CUSTOMREQUEST, $method);
		
	//Mise en place du xKey et des options
	curl_setopt($req, CURLOPT_HTTPHEADER, array(
	'X-Key: ' . $xKey,
	'Content-Type: application/json',
	'Content-Length: ' . strlen($message)));
	curl_setopt($req, CURLOPT_POSTFIELDS, $message);

	//Execution de la requete
	$result = curl_exec($req);

	//Verification des reponses
	$info = curl_getinfo($req);
	
	return (array('result' => $result, 'info' => $info, 'req' => $req));
}

?>

File: java/createValueList.java
import java.io.IOException;
import java.io.OutputStreamWriter;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

public class createValueList 
{
	public static void main(String[] args) throws IOException, JSONException, InterruptedException
	{
		//Ici, renseignez la xKey et les parametres personnalises
		String xKey = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
		String valueListsId = "012";	//Id de la liste a modifier ('null' si la liste doit etre creer)

		String name = "createValueList (java)";	//Nom de la liste
		Boolean ordered = false;	//Ordonnee : oui = 'true' / non = 'false'
		
		JSONArray values = new JSONArray();	//Valeurs de la liste
			JSONObject value1 = new JSONObject();	//Premiere valeur
				value1.put("index", 1);	//Numeros de la valeur
				value1.put("value", "Mr");	//Valeur du champs
			JSONObject value2 = new JSONObject();	//Premiere valeur
				value2.put("index", 2);	//Numeros de la valeur
				value2.put("value", "Mme");	//Valeur du champs
			values.put(value1);
			values.put(value2);
		
			
		//On trouve l'adresse pour la requete
		String url = "https://backoffice.mailperformance.com/valueLists/";
		if (valueListsId != null)
		{
			url = "https://backoffice.mailperformance.com/valueLists/" + valueListsId;
		}
		
		//Creation du Json du message
		JSONObject jsonData = new JSONObject();
		jsonData.put("name", name);
		jsonData.put("ordered", ordered);
		jsonData.put("values", values);
		
		//On affiche le message
		System.out.print(jsonData + "\n\n");
		
		//Lancement de la connexion
		HttpURLConnection con = null;
		if (valueListsId != null)
		{
			con = connection(url, xKey, jsonData, "PUT");
		}
		else
		{
			con = connection(url, xKey, jsonData, "POST");
		}
		
		//Verification des reponses
		int responseCode = con.getResponseCode();
		if (responseCode != 200)
		{
			//Affichage de l'erreur
			System.out.print("Error : " + responseCode + " " + con.getResponseMessage());
		}
		else
		{
			//La liste a bien ete creee
			System.out.print("The value list  : " + name + " is created / update.");
		}
		con.disconnect();
	}

	
	//Fonctions ----

	
	//Ouverture de la connexion
	public static HttpURLConnection openConn(String url, String xKey) throws MalformedURLException, IOException
	{
		//Lancement de la connexion
		HttpURLConnection con = (HttpURLConnection)new URL(url).openConnection();
		
		//Mise en place du xKey et des options
		con.setRequestProperty("X-Key", xKey);
		con.setRequestProperty("Content-Type", "application/json");
		return (con);
	}
	
	//Fonction de connexion et envoi des informations
	public static HttpURLConnection connection(String url, String xKey, JSONObject jsonMessage, String method) throws IOException
	{
		//Lancement de la connexion pour remplir la requete
		HttpURLConnection con = openConn(url, xKey);
		con.setRequestProperty("Content-Length", Integer.toString(jsonMessage.length()));
		con.setRequestMethod(method);
		con.setDoOutput(true);
				        
		//Envoi des informations dans la connexion
		OutputStreamWriter sendMessage = new OutputStreamWriter(con.getOutputStream());
		sendMessage.write(jsonMessage.toString());
		sendMessage.flush();
		sendMessage.close();
		return (con);
	}
}
